### DISCUSSION TITLE
SAM arguments (sam_kwargs): explanation

### DISCUSSION THREAD

**Discussion by aloboa** [opengeos/segment-geospatial#249] [Q&A] [answered]
Where can I find documentation on sam arguments, so that I can fine-tune the segmentation?
sam_kwargs = {
    "points_per_side": 32,
    "pred_iou_thresh": 0.86,
    "stability_score_thresh": 0.92,
    "crop_n_layers": 1,
    "crop_n_points_downscale_factor": 2,
    "min_mask_region_area": 100,
}

**Comment by giswqs** [comment, marked as answer]
It originates from the SAM model. See https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/automatic_mask_generator.py#L61

**Reply by aloboa** [reply]
Even with the definition, some are difficult to understand (and thus their impact on the result):
"pred_iou_thresh"
 "stability_score_thresh":
"crop_n_layers":

Although I'm running a set of tests with different parameters on a very small subscene to try to understand empirically, I would appreciate a pointer to an explanation.
